src/controller/CHIPDeviceController.cpp

```diff
@@ -2777,6 +2794,187 @@ CHIP_ERROR DeviceCommissioner::ParseICDInfo(ReadCommissioningInfo & info)
+                            return CHIP_ERROR_KEY_NOT_FOUND;
+                        }
+                        rootKeySpan                      = fabricDescriptor.rootPublicKey;
+                        info.JFAdminFabricTable.vendorId = fabricDescriptor.vendorID;
+                        info.JFAdminFabricTable.fabricId = fabricDescriptor.fabricID;
+
+                        if (fabricDescriptor.VIDVerificationStatement.HasValue())
+                        {
+                            ChipLogError(Controller, "JCM: Per-home RCAC are not supported by JF for now!");
+                            return CHIP_ERROR_CANCELLED;
+                        }
+                        ChipLogProgress(Controller, "JCM: Successfully parsed the Administrator Fabric Table");
+                        break;
+                    }
+                }
+                return CHIP_NO_ERROR;
+            }
+            case NOCs::Id: {
+                NOCs::TypeInfo::DecodableType nocs;
+                ReturnErrorOnFailure(this->mAttributeCache->Get<NOCs::TypeInfo>(path, nocs));
+
+                auto iter = nocs.begin();
+                while (iter.Next())
+                {
+                    auto & nocStruct = iter.GetValue();
+
+                    if (nocStruct.fabricIndex == info.JFAdministratorFabricIndex)
+                    {
+                        info.JFAdminNOC = nocStruct.noc;
+
+                        if (!nocStruct.icac.IsNull())
+                        {
+                            info.JFAdminICAC = nocStruct.icac.Value();
+                        }
+                        else
+                        {
+                            ChipLogError(Controller, "JCM: ICAC not present!");
+                            return CHIP_ERROR_CERT_NOT_FOUND;
+                        }
+                        ChipLogProgress(Controller, "JCM: Successfully parsed the Administrator NOC and ICAC");
+                        break;
+                    }
+                }
+                return CHIP_NO_ERROR;
+            }
+            default:
+                return CHIP_NO_ERROR;
+            }
+
+            return CHIP_NO_ERROR;
+        });
+
+    if (err != CHIP_NO_ERROR)
+    {
+        info.JFAdministratorFabricIndex = kUndefinedFabricIndex;
+        return err;
+    }
+
+    err = mAttributeCache->ForEachAttribute(
+        OperationalCredentials::Id, [this, &info, &rootKeySpan](const ConcreteAttributePath & path) {
+            using namespace chip::app::Clusters::OperationalCredentials::Attributes;
+            bool foundMatchingRcac = false;
+
+            switch (path.mAttributeId)
+            {
+            case TrustedRootCertificates::Id: {
+                TrustedRootCertificates::TypeInfo::DecodableType trustedCAs;
+                ReturnErrorOnFailure(this->mAttributeCache->Get<TrustedRootCertificates::TypeInfo>(path, trustedCAs));
+
+                auto iter = trustedCAs.begin();
+                while (iter.Next())
```

Comment:
All these iterators need status checks at end of iteration.

Reply:
Pls see my answer above.